app/Http/Controllers/Management/RoleController.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Http\Controllers\Management;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Repositories\Role\RoleInterface;
+use App\Repositories\Permission\PermissionInterface;
+use App\Models\Role;
+use App\Models\Permission;
+use App\Http\Requests\RoleRequest;
+
+class RoleController extends Controller
+{
+    protected $roleRepository;
+    protected $permissionRepository;
+    public function __construct(RoleInterface $roleRepository, PermissionInterface $permissionRepository)
+    {
```

Comment:
$this->source = $request->source, thử xem không phải mỗi action lại truyền source

Reply:
không truyền đc. source là parameter của route mà